client/rest-high-level/src/main/java/org/elasticsearch/client/MigrationClient.java

```diff
@@ -52,4 +56,20 @@ public IndexUpgradeInfoResponse getAssistance(IndexUpgradeInfoRequest request, R
         return restHighLevelClient.performRequestAndParseEntity(request, MigrationRequestConverters::getMigrationAssistance, options,
             IndexUpgradeInfoResponse::fromXContent, Collections.emptySet());
     }
+
+    public BulkByScrollResponse upgrade(IndexUpgradeRequest request, RequestOptions options) throws IOException {
+        return restHighLevelClient.performRequestAndParseEntity(request, MigrationRequestConverters::migrate, options,
+            BulkByScrollResponse::fromXContent, Collections.emptySet());
+    }
+
+    public IndexUpgradeSubmissionResponse submitUpgradeTask(IndexUpgradeRequest request, RequestOptions options) throws IOException {
```

Comment:
as discussed, I want to refactor that later to return TaskSubmissionResponse and use across other methods where wait_for_completion=false was used

Reply:
++.

Reply:
there is another PR where we discussed if it wouldn't be better to return just TaskID:
https://github.com/elastic/elasticsearch/pull/35202/files
I might get back to this one and refactor to this TaskID